course_info/tests/jasmine/controllers/DetailsControllerSpec.js

```diff
@@ -559,4 +566,161 @@ describe('Unit testing DetailsController', function () {
             expect(result).toEqual('ABC');
         });
     });
+
+    describe('associateNewSiteHandleKey', function(){
+        it('should call associateNewSite when a user hits the enter key', function(){
+            spyOn(dc, 'validNewSiteURL').and.returnValue(true);
+            spyOn(dc, 'associateNewSite');
+            var aEvent = angular.element.Event('keydown');
+            aEvent.keyCode = 13;
+            $("input").trigger(aEvent);
+            dc.associateNewSiteHandleKey(aEvent);
+            expect(dc.validNewSiteURL).toHaveBeenCalled();
+            expect(dc.associateNewSite).toHaveBeenCalled();
+
+        });
+
+        it('should not call associateNewSite when a user enteres an invalid url', function(){
+            spyOn(dc, 'validNewSiteURL').and.returnValue(false);
+            spyOn(dc, 'associateNewSite');
+            var aEvent = angular.element.Event('keydown');
+            aEvent.keyCode = 13;
+            $("input").trigger(aEvent);
+            dc.associateNewSiteHandleKey(aEvent);
+            expect(dc.validNewSiteURL).toHaveBeenCalled();
+            expect(dc.associateNewSite).not.toHaveBeenCalled();
+        });
+
+    });
+
+    describe('associateNewSite', function(){
+        beforeEach(function () {
+            var ci = {
+                course_instance_id: $routeParams.courseInstanceId,
+                title: 'Test Course Title',
+                short_title: 'Test',
+                sub_title: 'Jasime is your friend',
+                description: '<p>hello</p>',
+                sync_to_canvas : 1,
+                exclude_from_isites:0,
+                exclude_from_catalog:0,
+                sites: [
+                    {
+                        external_id: 'https://x.y.z/888',
+                        site_id: '888',
+                        map_type: 'official'
+                    },
+                    {
+                        external_id: 'https://x.y.z/999',
+                        site_id: '999',
+                        map_type: 'unofficial'
```

Comment:
These look different from the sites in the `dissociateSite`, should they be consistent?